File: 667.beautiful-arrangement-ii.cpp
/*
 * @lc app=leetcode id=667 lang=cpp
 *
 * [667] Beautiful Arrangement II
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<int> constructArray(int n, int k)
    {
        vector<int> res(n + 1);
        if (k > 1)
        {
            for (int i = 0; i < 2 * k && i < n; i += 2)
            {
                res[i] = i / 2 + 1;
                res[i + 1] = k + 2 - res[i];
            }
            for (int i = k + 1; i != n; ++i)
            {
                res[i] = i + 1;
            }
        }
        else
        {
            for (int i = 0; i != n; ++i)
            {
                res[i] = i + 1;
            }
        }
        res.pop_back();
        return res;
    }
};
// @lc code=end

File: py/667.beautiful-arrangement-ii.py
#
# @lc app=leetcode id=667 lang=python3
#
# [667] Beautiful Arrangement II
#

# @lc code=start
class Solution:
    def constructArray(self, n: int, k: int) -> List[int]:
        if k > 1:
            res = [None]*(n+1)
            for i in range(0, min(2*k, n), 2):
                res[i] = i//2+1
                res[i+1] = k+2-res[i]
            for i in range(k+1, n):
                res[i] = i+1
            res.pop()
        else:
            res = list(range(1, n+1))
        return res


# @lc code=end
